compiler/ast/parser.nim

```diff
@@ -666,101 +670,57 @@ proc identOrLiteral(p: var Parser, mode: PrimaryMode): PNode =
-    result = newIntNodeP(nkUInt16Lit, p.tok.iNumber, p)
-    setBaseFlags(result, p.tok.base)
-    p.getTok
-  of tkUInt32Lit:
-    result = newIntNodeP(nkUInt32Lit, p.tok.iNumber, p)
-    setBaseFlags(result, p.tok.base)
-    p.getTok
-  of tkUInt64Lit:
-    result = newIntNodeP(nkUInt64Lit, p.tok.iNumber, p)
-    setBaseFlags(result, p.tok.base)
-    p.getTok
-  of tkFloatLit:
-    result = newFloatNodeP(nkFloatLit, p.tok.fNumber, p)
-    setBaseFlags(result, p.tok.base)
-    p.getTok
-  of tkFloat32Lit:
-    result = newFloatNodeP(nkFloat32Lit, p.tok.fNumber, p)
-    setBaseFlags(result, p.tok.base)
-    p.getTok
-  of tkFloat64Lit:
-    result = newFloatNodeP(nkFloat64Lit, p.tok.fNumber, p)
-    setBaseFlags(result, p.tok.base)
-    p.getTok
-  of tkFloat128Lit:
-    result = newFloatNodeP(nkFloat128Lit, p.tok.fNumber, p)
-    setBaseFlags(result, p.tok.base)
-    p.getTok
-  of tkStrLit:
-    result = newStrNodeP(nkStrLit, p.tok.literal, p)
-    p.getTok
-  of tkRStrLit:
-    result = newStrNodeP(nkRStrLit, p.tok.literal, p)
-    p.getTok
-  of tkTripleStrLit:
-    result = newStrNodeP(nkTripleStrLit, p.tok.literal, p)
-    p.getTok
-  of tkCharLit:
-    result = newIntNodeP(nkCharLit, ord(p.tok.literal[0]), p)
-    p.getTok
+  of tkIntLit:       result = p.newNode(nkIntLit,       p.tok); p.getTok
+  of tkInt8Lit:      result = p.newNode(nkInt8Lit,      p.tok); p.getTok
+  of tkInt16Lit:     result = p.newNode(nkInt16Lit,     p.tok); p.getTok
+  of tkInt32Lit:     result = p.newNode(nkInt32Lit,     p.tok); p.getTok
+  of tkInt64Lit:     result = p.newNode(nkInt64Lit,     p.tok); p.getTok
+  of tkUIntLit:      result = p.newNode(nkUIntLit,      p.tok); p.getTok
+  of tkUInt8Lit:     result = p.newNode(nkUInt8Lit,     p.tok); p.getTok
+  of tkUInt16Lit:    result = p.newNode(nkUInt16Lit,    p.tok); p.getTok
+  of tkUInt32Lit:    result = p.newNode(nkUInt32Lit,    p.tok); p.getTok
+  of tkUInt64Lit:    result = p.newNode(nkUInt64Lit,    p.tok); p.getTok
+  of tkFloatLit:     result = p.newNode(nkFloatLit,     p.tok); p.getTok
+  of tkFloat32Lit:   result = p.newNode(nkFloat32Lit,   p.tok); p.getTok
+  of tkFloat64Lit:   result = p.newNode(nkFloat64Lit,   p.tok); p.getTok
+  of tkFloat128Lit:  result = p.newNode(nkFloat128Lit,  p.tok); p.getTok
+  of tkStrLit:       result = p.newNode(nkStrLit,       p.tok); p.getTok
+  of tkRStrLit:      result = p.newNode(nkRStrLit,      p.tok); p.getTok
+  of tkTripleStrLit: result = p.newNode(nkTripleStrLit, p.tok); p.getTok
+  of tkCharLit:      result = p.newNode(nkCharLit,      p.tok); p.getTok
   of tkCustomLit:
     let splitPos = p.tok.iNumber.int
-    let str = newStrNodeP(nkRStrLit, p.tok.literal.substr(0, splitPos-1), p)
-    let callee = newIdentNodeP(getIdent(p.lex.cache, p.tok.literal.substr(splitPos)), p)
-    result = newTreeI(nkDotExpr, p.lineInfo, [str, callee])
+    var tok = p.tok
+    tok.literal = p.tok.literal.substr(0, splitPos - 1)
+    let str = p.newNode(nkRStrLit, tok)
```

Comment:
Maybe I'm totally off base here, but I don't think the before and after are equivalent, in the former code we're substring-ing on the same literal. While in the latter we're doing a sub string on the result of the first to get the second?

Reply:
Oh maybe it's a value type ... checking

Reply:
Yup, value type.

Reply:
```
str = thing[0, splitPos - 1]
callee = thing[splitPos]
result = dotExpr[str, callee]
```

In both cases